### ISSUE TITLE
ServiceProxy(): Forces JSON strings to be url decoded

### ISSUE THREAD

**Issue by cyberdqueued** [cenobites/flask-jsonrpc#25]
Currently *ServiceProxy()* uses *urllib*, which in Python 2.x doesn't allow for HTTP header modification. This becomes an issue when trying to send JSON strings with characters that contain values that *look* like url-encoded characters. By default *urllib* sends **POST** data with the HTTP header "Content-Type" set to "application/x-www-form-urlencoded". The JSON client should be using "application/json" for the "Content-Type" in the HTTP header. The following code is meant to demonstrate the problem:

Server:
```
from flask import Flask
from flask_jsonrpc import JSONRPC
app = Flask("DEBUG")
annote = JSONRPC(app,'/DEBUG',enable_web_browsable_api=True)
@annote.method('one(str)->str')
def one(instr):
    print("string is %s" % instr)
    return instr

app.run(host='0.0.0.0',port=5001,debug=True)
```

Client:
```
from flask_jsonrpc.proxy import ServiceProxy
s=ServiceProxy('http://0.0.0.0:5001/DEBUG')
print(s.one('No%20Space'))
```

Resulting output:
```
{u'jsonrpc': u'2.0', u'id': u'df4932b0-8551-11e4-86b1-0026b9641702', u'result': u'No Space '}
```

The "%20" was decoded into a space (" ").

The solution for this would be to replace *urllib* with *urllib2* in _compat.py and subsequently the import of urllib in *ServiceProxy()*. The default HTTP header "Content-Type" should be set to "application/json" but there will be a overriding parameter available so users can provide their own HTTP headers dictionary.

The following are code snippets that illustrate the solution:

**_compat.py:**

For PY3:
```
encode_filename = _identity
    get_next = lambda x: x.__next__

    from urllib.parse import urlparse
    from urllib.request import urlopen
    from urllib.request import Request
```
For PY2:
```
        return filename

    from urlparse import urlparse
    from urllib2 import urlopen
    from urllib2 import Request
```

**proxy.py:**
```
class ServiceProxy(object):

    DEFAULT_HEADERS = {"Content-Type":"application/json"}
    
    def __init__(self, service_url, service_name=None, version='2.0', headers=None):
        self.headers = headers or self.DEFAULT_HEADERS
        self.version = text_type(version)
...
...
...
    def send_payload(self, params):
        """Performs the actual sending action and returns the result
        """
        data = json.dumps({
            'jsonrpc': self.version,
            'method': self.service_name,
            'params': params,
            'id': text_type(uuid.uuid1())
        })
        data_binary = data.encode('utf-8')
        url_request = Request(self.service_url,data_binary,headers=self.headers)
        return urlopen(url_request).read()
```

Thanks for taking a look!

**Comment by nycholas** [comment]
Hello,

I will integrate in new release.

Thank you.

**Comment by nycholas** [comment]
Fixed in version 0.3.
